sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -141,12 +150,21 @@ def fit(self, X, y):
             if not (self._is_fitted() and self.warm_start):
                 self._train_val_split_seed = rng.randint(1024)
 
-            X_train, X_val, y_train, y_val = train_test_split(
-                X, y, test_size=self.validation_fraction, stratify=stratify,
-                random_state=self._train_val_split_seed)
+            if sample_weight is None:
+                X_train, X_val, y_train, y_val = train_test_split(
+                    X, y, test_size=self.validation_fraction,
+                    stratify=stratify,
+                    random_state=self._train_val_split_seed)
+                sample_weight_train = sample_weight_val = None
+            else:
+                (X_train, X_val, y_train, y_val, sample_weight_train,
```

Comment:
I think we want to subsample based on the weights here too, since the data is going to be used for training? I'm not sure.

Reply:
I think that goes in the same category as `resample` in #14692. I'd argue that it'd be a fix for `train_test_split`, and beyond the scope of this PR?

Reply:
Probably yeah. In any case we'd need a TODO for us to remember